components/util/remutex.rs

```diff
@@ -0,0 +1,217 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Oh, I thought you copied it straight from libstd, but yeah, the tests still apply.